src/cargo/ops/resolve.rs

```diff
@@ -243,33 +261,43 @@ pub fn resolve_ws_with_opts<'gctx>(
         force_all_targets,
     )?;
 
-    let feature_opts = FeatureOpts::new(ws, has_dev_units, force_all_targets)?;
-    let resolved_features = FeatureResolver::resolve(
-        ws,
-        target_data,
-        &resolved_with_overrides,
-        &pkg_set,
-        cli_features,
-        specs,
-        requested_targets,
-        feature_opts,
-    )?;
+    let mut specs_and_features = Vec::new();
 
-    pkg_set.warn_no_lib_packages_and_artifact_libs_overlapping_deps(
-        ws,
-        &resolved_with_overrides,
-        &member_ids,
-        has_dev_units,
-        requested_targets,
-        target_data,
-        force_all_targets,
-    )?;
+    for specs in individual_specs {
```

Comment:
So we're creating independent resolves for each package.  I take it we can then leave merging of these graphs to a future possibility to further optimize this?

Reply:
Just independent `ResolvedFeatures`, it seems everything works with just a single `targeted_resolve`, though I did not expect that since the field is documeted with "The narrowed resolve, with the specific features enabled." Then from those independent unit graphs are constructed and those are then merged into a single build context.

I probably don't understand the whole process well enough, but I don't see a way to merge anything earlier?

Reply:
Maybe phrased differently, if you have the following
```
a -> common w/ features=foo
b -> common w/ features=foo
c -> common
```
Will we have 2 or 3 entries in the unit graph?

Reply:
It should have 2 entries for `common` (so 5 entries in total).

I've added a test where there are the same features enabled on a common dependency and it is compiled just once.